### PR TITLE
Fix dockerfile.registry to work with origin-operator-registry 4.2.0

### PR DESCRIPTION
**Pull request by ksimon1**
Latest build of registry container fails with error: 
time="2019-08-19T10:13:44Z" level=fatal msg="permissive mode disabled" error="error loading manifests from directory: could not decode contents of file /registry/bundles.db into package: error converting YAML to JSON: yaml: control characters are not allowed"

This PR fixes this error (same issue as here: https://github.com/kubevirt/cluster-network-addons-operator/pull/215).

/cc @MarSik, @fromanirh

### CODE REVIEW

**Comment by MarSik** [comment]
The fix itself looks good, but I am a bit uncertain about the maintainability. Do we expect to need different manifest bundle for different openshift versions? In other words, should this be configurable somehow?

File: build/Dockerfile.registry [hunk 1]
```diff
@@ -1,10 +1,13 @@
-FROM quay.io/openshift/origin-operator-registry
+FROM quay.io/openshift/origin-operator-registry:4.2.0
 
-COPY manifests /registry
+COPY manifests manifests
```
Comment (fromanirh) [inline build/Dockerfile.registry]: I'm not sure thus works OK with HCO/registry bundles. The filesystem layout is significant here.

File: build/Dockerfile.registry [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+
+# Bundle should include only manifests related to the CSV,
+# drop all additional files.
+RUN find manifests/kubevirt-ssp-operator/*/ -type f ! -name '*_crd.*' ! -name '*.clusterserviceversion.*' -exec rm -f {} \;
```
Comment (fromanirh) [inline build/Dockerfile.registry]: this works but I'm not too happy with it. We should log spurious/unexpected entries.

File: build/Dockerfile.registry [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
 
 # Initialize the database
-RUN rm -rf /registry/generated && \
-    initializer --manifests /registry --output bundles.db
+RUN initializer --manifests manifests/kubevirt-ssp-operator/ --output bundles.db
```
Comment (fromanirh) [inline build/Dockerfile.registry]: same concern as per line #3

File: build/Dockerfile.registry (hunk 1 above)
Comment (fromanirh) [reply build/Dockerfile.registry]: My source is just https://github.com/operator-framework/operator-registry

File: build/Dockerfile.registry (hunk 2 above)
Comment (fromanirh) [reply build/Dockerfile.registry]: maybe just use the `-v flag` of the `rm` command?

File: build/Dockerfile.registry (hunk 1 above)
Comment (ksimon1) [reply build/Dockerfile.registry]: according to this https://github.com/operator-framework/operator-registry/blob/master/cmd/registry-server/main.go#L36 registry server cares only about bundles.db path

**Pull request merged**